<h1 style='font-size:40px'> 
    <a href='https://otexts.com/fpp2/residuals.html'>Residuals</a></h1>
<div> 
    <ul style='font-size:20px'> 
        <li> 
            Os resíduos de um modelo seriam a diferença entre o valor real da série e a previsão. 
        </li>
    </ul>
 </div>

 <center style='font-size:20px;margin-top:20px'> 
        <figure>
            $e_{t}=y_{t}-\hat{y}_{t}$
            <figcaption style='font-size:15px'> 
                Resíduo de um time-step $t$
            </figcaption>
        </figure>
    </center>

<div> 
    <ul style='font-size:20px'> 
        <li> 
            Quando pensamos em um modelo de forecasting, esperamos que os seus resíduos tenham as seguintes características:
            <ol> 
                <li>
                    <i>Ausência de Autocorrelação:</i> Caso o contrário entende-se que parte da informação da série não foi capturada pelo modelo.
                </li>
                <li>
                    <i>Média zero:</i> Indicando ausência de bias.
                </li>
            </ol>
        </li>
        <li>
            Pensando na computação de Intervalos de Confiança, é recomendado também que os resíduos tenham <i> Variância Constante</i> e <i> Distribuição Normal</i>. 
        </li>
    </ul>
 </div>

<h2 style='font-size:30px'> Testes para Resíduos</h2>
<div> 
    <ul style='font-size:20px'> 
        <li> 
            Os estatísticos usam alguns testes que buscam dar uma quantia de se uma dada série de resíduos é autocorrelacionada ou não.
        </li>
        <li>
            Entre os mais famosos, podemos mencionar o <strong> Box-Pierce Test</strong> e o <strong> Ljung-Box Test</strong>.
        </li>
        <li>
            Seja $r_{k}$ a autocorrelação para o lag de $k$ time-steps, $l$ o máximo de time-steps sendo avaliado e $T$ o tamanho da série, esses testes podem ser definidos como:
        </li>
    </ul>
</div>

 <center style='font-size:20px;margin-top:20px'> 
        <figure>
            $\displaystyle Q=T\sum_{k=1}^{l}{r_{k}^{2}}$
            <figcaption style='font-size:15px'>
                Estatística do Box-Pierce Test
            </figcaption>
        </figure>
    </center>

 <center style='font-size:20px;margin-top:20px'> 
        <figure>
            $\displaystyle Q^{*}=T(T+2)\sum_{k=1}^{l}{(T-k)^{-1}r_{k}^{2}}$
            <figcaption style='font-size:15px'>
                Estatística do Ljung-Box Test
            </figcaption>
        </figure>
    </center>

<h2 style='font-size:30px'> Resíduos x Erros</h2>
<div> 
    <ul style='font-size:20px'> 
        <li> 
            O site descreve que usamos o termo "resíduo" e "erro" em contextos distintos.
        </li>
        <li>
            Devemos usar o termo "resíduo" para apenas a diferença $y_{t}-\hat{y}_{t}$ do set de treino, enquanto que "erro" para essa diferença no set de teste.
        </li>
    </ul>
</div>

<h2 style='font-size:30px'> Time Series Cross-Validation</h2>
<div> 
    <ul style='font-size:20px'> 
        <li> 
            A Cross-Validation de séries temporais é uma forma de avaliar de maneira mais robusta a qualidade de um algoritmo.
        </li>
        <li>
            Dada uma série de tamanho $T$, treinamos o modelo de $y_{1},\ldots,y_{n}$, e validamos os resíduos de $y_{n+1},\ldots,y_{n+1+k}$, sendo $k$ o tamanho da janela de inferência desejada.     
        </li>
        <li>
            Repetimos esse processo para diferentes valores de $n$.
        </li>
    </ul>
</div>

 <center style='font-size:20px;margin-top:20px'> 
        <figure>
            <img src='../img/cross-validation.png'>
            <figcaption style='font-size:15px'>
                Ilustração da Cross-Validation. Observe que ela nos dá uma amostragem muito mais significativa sobre a performance do modelo.
            </figcaption>
        </figure>
    </center>

In [5]:
! git add .
! git commit -am 'Continuar de "Multi-step prediction intervals"'
! git push

[master d7997a4] Continuar de seção "3.4/Time series cross-validation"
 2 files changed, 45 insertions(+), 12 deletions(-)
Enumerating objects: 9, done.
Counting objects: 100% (9/9), done.
Delta compression using up to 24 threads
Compressing objects: 100% (5/5), done.
Writing objects: 100% (5/5), 956 bytes | 956.00 KiB/s, done.
Total 5 (delta 4), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (4/4), completed with 4 local objects.
To https://github.com/felipesveiga/time-series.git
   cd1ed81..d7997a4  master -> master


<p style='color:red'> Vi "3.4"; Iniciei seção "3.5"; Continuar de "Multi-step prediction intervals"